src/test/scala/uk/gov/nationalarchives/tdr/api/routes/ClientFileMetadataRouteSpec.scala

```diff
@@ -0,0 +1,85 @@
+                                fileId: Option[Long] = None,
+                                originalPath: Option[String] = None,
+                                checksum: Option[String] = None,
+                                checksumType: Option[String] = None,
+                                lastModified: Option[Long] = None,
+                                createdDate: Option[Long] = None,
+                                fileSize: Option[BigDecimal] = None,
+                                datetime: Option[Long] = None,
+                                clientFileMetadataId: Option[Long] = None
+                              )
+  case class AddClientFileMetadata(addClientFileMetadata: ClientFileMetadata) extends TestRequest
+
+  override def beforeEach(): Unit = {
+    resetDatabase()
+  }
+
+  val runTestMutation: (String, OAuth2BearerToken) => GraphqlMutationData =
+    runTestRequest[GraphqlMutationData](addClientFileMetadataJsonFilePrefix)
+  val expectedMutationResponse: String => GraphqlMutationData =
+    getDataFromFile[GraphqlMutationData](addClientFileMetadataJsonFilePrefix)
+
+  "The api" should "return all requested fields from inserted Client File metadata object" in {
+
+    val expectedResponse: GraphqlMutationData = expectedMutationResponse("data_all")
+    val response: GraphqlMutationData = runTestMutation("mutation_alldata", validUserToken())
+    response.data.get.addClientFileMetadata should equal(expectedResponse.data.get.addClientFileMetadata)
+
+    checkClientFileMetadataExists(response.data.get.addClientFileMetadata.clientFileMetadataId.get)
+  }
+
+  "The api" should "return the expected data from inserted Client File metadata object" in {
+
+    val expectedResponse: GraphqlMutationData = expectedMutationResponse("data_some")
+    val response: GraphqlMutationData = runTestMutation("mutation_somedata", validUserToken())
+    response.data.get.addClientFileMetadata should equal(expectedResponse.data.get.addClientFileMetadata)
+
+    checkClientFileMetadataExists(response.data.get.addClientFileMetadata.clientFileMetadataId.get)
+  }
+
+  "The api" should "throw an error if the file id field is not provided" in {
+
+    val expectedResponse: GraphqlMutationData = expectedMutationResponse("data_fileid_missing")
+    val response: GraphqlMutationData = runTestMutation("mutation_missingfileid", validUserToken())
+
+    response.errors.head.message should equal (expectedResponse.errors.head.message)
+  }
+
+  private def checkClientFileMetadataExists(clientFileMetadataId: Long): Unit = {
+    val sql = s"select * from consignmentapi.ClientFileMetadata where ClientFileMetadataId = $clientFileMetadataId;"
+    val ps: PreparedStatement = DbConnection.db.source.createConnection().prepareStatement(sql)
+    val rs: ResultSet = ps.executeQuery()
+    rs.next()
+    rs.getString("ClientFileMetadataId") should equal(clientFileMetadataId.toString)
+  }
+
+  private def resetDatabase(): Unit = {
```

Comment:
Are we doing DB resets in a lot of places now, and would it be easier to share the teardown by wiping the DB after every test?

Reply:
There are a lot of resets, but I'm not sure having a sharedown is a good idea. From a personal point of view with unit tests I prefer everything in one place. I've seen having a shared functionality between tests causing unintended consequences across unit tests

Reply:
To be honest, it would be nice if we could create and tear down the entire DB for each test (and not have many integration tests, because that's slow). I'm definitely not suggesting that we have tests that depend on setup from previous tests! 🙂 

I'm actually doing some API dev at the moment, so it's nice to get a feel for how easy the tests are to work with. Let's leave it like this for now. I might try refactoring things a bit if anything seem fiddly to work with, and then see what you all think.